File: libraries/hds/cmp_len.f
      subroutine cmp_len(struct, comp, len, status)
*+
*  Name:
*     CMP_LEN

*  Purpose:
*     Component String Precision enquiry.

*  Language:
*     VAX Fortran

*  Invocation:
*     CALL CMP_LEN(LOC, NAME, LEN, STATUS)

*  Description:
*     A string precision enquiry is made for a structure component
*     of type '_CHAR*LEN'.

*  Arguments:
*     LOC=CHARACTER*(DAT__SZLOC)
*        Variable containing a locator associated with a structured
*        data object.
*     NAME=CHARACTER*(*)
*        Expression specifying the component name of a primitive
*        object contained in the structure.
*     LEN=INTEGER
*        Variable to receive number of characters per value.
*     STATUS=INTEGER
*        Variable holding the status value. If this variable is not
*        SAI__OK on input, the routine will return without action.
*        If the routine fails to complete, this variable will be
*        set to an appropriate error number.

*  Algorithm:
*     Get object locator using DAT_FIND and then get size from
*     DAT_LEN.

*  Authors:
*     Jack Giddings (UCL::JRG)
*     {enter_new_authors_here}

*  History:
*     3-JAN-1983:  Original.  (UCL::JRG)
*     15-APR-1987:  Improved prologue layout (RAL::AJC)
*     {enter_further_changes_here}

*  Bugs:
*     {note_any_bugs_here}

*-

*  Global Constants:
      INCLUDE 'SAE_PAR'
      INCLUDE 'DAT_PAR'

*  Arguments Given:
      character*(*) struct		! Structure Locator
      character*(*) comp		! Component Name

*  Arguments Returned:
      integer len			! String precision of component
*    Status return :
      integer status			! Status Return

*  Local Variables:
      character*(DAT__SZLOC) loc	! Component locator

*.


      if (status .eq. SAI__OK) then
         call dat_find(struct, comp, loc, status)
         if (status .ne. SAI__OK) then
            call cmp_erdsn(struct, comp, status)
         else
            call dat_len(loc, len, status)
            if (status .ne. SAI__OK) then
               call cmp_erdsn(struct, comp, status)
            endif
            call dat_annul(loc, status)
         endif
      endif

      end

File: libraries/hds/cmpLen.c
#include "sae_par.h"
#include "hds.h"
#include "dat_par.h"

/*
*+
*  Name:
*     cmpLen

*  Purpose:
*     Component length enquiry.
*
*  Synopsis:
*     void cmpLen( HDSLoc *struc, const char *comp, size_t *len, int *status)

*  Description:
*     A length enquiry is made for a structure component.

*  Parameters:
*     struc
*        Locator associated with a structured data object.
*     comp
*        The component name of a primitive object contained in the structure.
*     len
*        The number of bytes per element.
*     status
*        Global status value.

*  Authors:
*     GSB: Graham Bell (EAO)

*  History:
*     07-MAY-2019 (GSB):
*        Original version, based on equivalent Fortran routine.
*-
*/

void cmpLen( HDSLoc *struc, const char *comp, size_t *len, int *status ) {
   HDSLoc *loc = NULL;
   if (*status != SAI__OK) {
      return;
   }
   datFind(struc, comp, &loc, status);
   datLen(loc, len, status);
   datAnnul(&loc, status);
}
